optima/programs.py

```diff
@@ -548,7 +548,8 @@ def getoutcomes(self, coverage=None, t=None, parset=None, results=None, sample='
                             if not self.covout[thispartype][thispop].ccopars[thisprog.short]:
                                 print('WARNING: no coverage-outcome parameters defined for program  "%s", population "%s" and parameter "%s". Skipping over... ' % (thisprog.short, thispop, thispartype))
                                 outcomes[thispartype][thispop] = None
-                            else: outcomes[thispartype][thispop] += thiscov[thisprog.short]*delta[thisprog.short]
+                            else: 
+                                outcomes[thispartype][thispop] += thiscov[thisprog.short]*delta[thisprog.short] * self.programs[thisprog.short].costcovfn.getccopar(t=t, sample=sample)['saturation']
```

Comment:
@cliffckerr I don't think this is right. Saturation is supposed to enter in via the cost-coverage function, so it should already have been taken into account and shouldn't need to come in again via the coverage-outcome function. I ran `testmodalities` and the test there doesn't pass with this line here (it does pass if I revert to the previous line).

Reply:
@robynstuart perhaps the fix isn't correct, but the current version is _definitely_ wrong: see `analyses/debug/modalities.py`, which demonstrates that if you reduce the saturation, you _increase_ the impact of the program.